pytorch_blade/src/torch-mlir/lib/Dialect/TorchConversion/Transforms/ReduceTensorConversions.cpp

```diff
@@ -0,0 +1,107 @@
+// Copyright 2022 The BladeDISC Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "mlir/Transforms/DialectConversion.h"
+
+#include "torch-mlir/Conversion/MhloPasses.h"
+#include "torch-mlir/Conversion/TorchToMhlo/TorchToMhlo.h"
+#include "torch-mlir/Dialect/Torch/IR/TorchDialect.h"
+#include "torch-mlir/Dialect/Torch/IR/TorchOps.h"
+#include "torch-mlir/Dialect/Torch/IR/TorchTypes.h"
+#include "torch-mlir/Dialect/TorchConversion/IR/TorchConversionOps.h"
+
+using namespace mlir;
+using namespace mlir::torch;
+using namespace mlir::torch::Torch;
+using namespace mlir::torch::TorchConversion;
+
+namespace {
+template <typename T>
+Type removeUser(Value arg, Type default_type) {
+  Type new_type = default_type;
+  for (auto user : arg.getUsers()) {
+    if (mlir::isa<T>(user)) {
+      auto val = user->getResult(0);
+      new_type = val.getType();
+      val.replaceAllUsesWith(arg);
+      user->erase();
+    }
+  }
+  return new_type;
+}
+
+template <typename T>
+Value backtraceOperand(Value operand) {
+  auto op = operand.getDefiningOp();
+  if (op && mlir::isa<T>(op)) {
```

Comment:
nit: What's the difference with this and above lines with `operand.getDefiningOp<T>()`?